### PR TITLE
fix: use angular/cli instead of angular-librarian

### LINKED ISSUE #57: "export 'FontAwesomeModule' (imported as 'i97') was not found in '@fortawesome/angular-fontawesome'
Did anyone get this issue when using angular-fontawesome with Angular 6?

screen shot 2018-06-21 at 12 05 05 pm

screen shot 2018-06-21 at 12 06 24 pm

screen shot 2018-06-21 at 12 08 41 pm

Adding this line of code `import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';` causes that error.

### PR DESCRIPTION
**Pull request by scttcper**
Currently, this project was generated by and uses a bunch of webpack scripts from angular-librarian. I don't think support there is good and packages here have become outdated as a result. This PR will put the demo application and library into [angular-cli](https://github.com/angular/angular-cli) which is fully supported.

All the commands are the same, the folders have moved slightly. Before `src` was just the library, now its in `src/lib` and the demo has moved to `src/app`

fixes #57  

notes:  
had to turn off tsconfig strict mode because the newer version of typescript has new strict rules that the code doesn't follow.

### CODE REVIEW

**Comment by mlwilkerson** [comment]
Well, I always did intend to use the official `angular-cli` tools when they eventually became available to handle library packaging. So the last _three_ library packaging solutions have been temporary concessions anyway, in my view.

Does it require one to use Angular 6 in order to get the library support built in with `angular-cli`?

@devoto13 What would you think of jumping into the official library packaging support with `angular-cli` instead of fixing up the build with Angular Librarian?

**Comment by scttcper** [comment]
building with angular 6 does not break compatibility with angular 5. Its a bit of a chore to use the latest angular-cli with older angular and older rxjs

**Comment by devoto13** [comment]
@mlwilkerson Totally fine by me! Especially since @scttcper have already done a great job and prepared a PR for it :+1:

File: package.json [hunk 1]
```diff
@@ -37,55 +37,48 @@
     "url": "https://github.com/FortAwesome/angular-fontawesome/issues"
   },
   "homepage": "https://github.com/FortAwesome/angular-fontawesome",
+  "dependencies": {
+    "@angular/animations": "^6.0.6",
```
Comment (devoto13) [inline package.json]: Maybe remove unnecessary dependencies from here: animations, forms, http and router?

File: src/lib/shared/models/props.model.ts [hunk 2]
```diff
@@ -23,9 +23,9 @@ export interface FaProps {
   counter?: boolean;
   inverse?: boolean;
   flip?: FlipProp;
-  size?: SizeProp;
-  pull?: PullProp;
-  rotate?: RotateProp;
+  size?: SizeProp | null;
```
Comment (devoto13) [inline src/lib/shared/models/props.model.ts]: Can you explain why this change is needed?

File: src/tsconfig.spec.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (devoto13) [inline src/tsconfig.spec.json]: I think you can remove `tsconfig.test.json` in the root as it is not used anymore.

File: tsconfig.json [hunk 4]
```diff
@@ -1,29 +1,21 @@
...
-    },
-    "files": [
-        "./src/index.ts"
+  "compileOnSave": false,
+  "compilerOptions": {
+    "baseUrl": "./",
+    "outDir": "./dist/out-tsc",
+    "sourceMap": true,
+    "declaration": false,
+    "strict": false,
```
Comment (devoto13) [inline tsconfig.json]: Can you change this back to true and disable specific strict features, which are not currently supported instead? E.g. `strict: true, strictPropertyInitialization: false`.

File: src/lib/shared/models/props.model.ts (hunk 2 above)
Comment (scttcper) [reply src/lib/shared/models/props.model.ts]: turning off strictNullChecks fixes it, reverted
